### ISSUE TITLE
KNITRO finds different solutions for some tests

### ISSUE THREAD

**Issue by frapac** [lanl-ansi/MINLPTests.jl#9]
I am currently testing the MOI branch of KNITRO.jl on [od/moi](https://github.com/lanl-ansi/MINLPTests.jl/tree/od/moi). The package is working well, and almost all tests pass! 

However, using the following Optimizer factory: 
```julia
solver = JuMP.with_optimizer(KNITRO.Optimizer, opttol=1e-10)
```
I get different results for some tests: 
- [nlp_005_010](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp/005_010.jl) (`1.5449723904763646` instead of `1.5449760741521967`)
- [nlp_mi_003_014](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp-mi/003_014.jl) and [nlp_mi_003_015](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp-mi/003_015.jl) (`11.0` instead of `11.000000198181866`)
- [nlp_mi_004_010](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp-mi/004_010.jl) and [nlp_mi_004_011](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp-mi/004_011.jl) (`-4.5768807316995614` instead of ` -4.576881309190901`)

What guarantee do we have on the value of the solutions? Is it worth to consider more than 8 digits for MIP problems?

**Comment by odow** [comment]
It probably makes sense to loosen the default tolerances to 1e-5. I'm not sure how @ccoffrin generated the answers to such precision.

**Comment by ccoffrin** [comment]
Agreed.  I think it is reasonable to relax the objective tolerance.  If `1e-5` is an easy fix I am fine with that.  However, a more sustainable strategy might be to configure approx. to focus on low relative error rather than tight absolute errors.

**Comment by odow** [comment]
KNITRO should just choose its own comparison tolerance when running the tests. It's easy to do: `test_nlp_mi(optimizer, primal_tol=1e-4)`

https://github.com/lanl-ansi/MINLPTests.jl/blob/1010de5df6a15bc3210b616318fb5bab168931b9/src/MINLPTests.jl#L98-L100

The bigger problem is that the solutions we currently test against aren't the analytic optimum.

https://github.com/lanl-ansi/MINLPTests.jl/blob/1010de5df6a15bc3210b616318fb5bab168931b9/src/nlp-mi/003_014.jl#L19

For example, this should be `11.0` exactly.

**Comment by odow** [comment]
@frapac, feel free to update any solutions you get to the tighter tolerances found by KNITRO by making a PR into my `od/moi` branch.

**Comment by odow** [comment]
Closed by #10
